### DISCUSSION TITLE
Linux (Debian based)

### DISCUSSION THREAD

**Discussion by qskousen** [Satellile/yara#10] [General]
Thank you for this tool! Just wanted to relate my experience compiling this for (Debian-based) Linux.

I needed to install a couple libraries:

`sudo apt install libxcb1-dev libxcb-xfixes0-dev`

(I may have had other needed libraries already installed.)

But once those were installed, it compiled fine and ran fine, at least for saving and loading queues. Trying to use the `image` command resulted in an error:

```
thread 'main' panicked at src/main.rs:556:53:
no entry found for key
```

I have not looked yet at what is causing that.

**Comment by qskousen** [comment]
I should also add that I compiled it with `1.81`.

**Comment by Satellile** [comment]
Thanks for sharing and letting me know!

For the crash, it looks it successfully reads embedded JSON data from the image file, but then fails when parsing the data. Unless perhaps `serde_json` operates differently on Linux vs Windows, I'm guessing the issue is independent of OS. Most likely, either custom nodes are breaking it (I recall not really building this too robustly to account for custom nodes), or ComfyUI changed around a bit (it's been awhile since I've since updated or used this myself).

For debugging, the first thing I'd do would be to add a line `dbg!(&field)` between lines 555 and 556 in main.rs, just to see which field it's trying to access when crashing, and then I might print the full JSON as well just to check it out, that would probably be enough to figure out the issue to then start a fix.

Sadly I prob won't personally fix it anytime soon since I'm pretty busy with another project, but I'll add this on my list of things to check out if I return to patch this program up.
